### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by emilybthorpe**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -117,7 +117,6 @@ $RECYCLE.BIN/
 *.lnk
 
 # End of https://www.toptal.com/developers/gitignore/api/linux,macos,windows,particle,c++
-docs/
```
Comment (ntlhui) [inline .gitignore:120]: Note that I am specifically ignoring docs for doxygen.  We can probably move doxygen output somewhere else.

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+- System to store data to onboard from sensors
+- Sleep until miliseconds function
+
+************System************
+
+- Main file that initalizes all sensors and components
+    - Initalized components:
+        - sensors: gps, imu, temperature, electric compass, water sensor, ICM20648 (6-axis MotionTracking device, 3-axis gyroscope, 3-axis accelerometer, and a Digital Motion Processor) , AK09916 (3-axis compass
+        - computer components: NVRAM, flash memory, charger, led (battery and water), and clock/timers, i2c
```
Comment (ntlhui) [inline docs/architecture.md:73]: Please stick with the `#` style headings

**Comment by ntlhui** [comment]
I would suggest a diagram like this:
```
@startuml Service Diagram
package "Device OS" {
    [Cellular] as cellular
    [I2C] as i2c
    [Serial] as serial
    [Filesystem] as fs
    [SPI] as spi
}


package "Application" {
    [CLI] as cli
    [Ride] as ride
}

package "Board Support Package" {
    [Temp Sensor] as temp_sensor
    [GPS] as gps
}
[ConIO] as conio
[Recorder] as recorder

ride --> temp_sensor
ride --> gps
ride --> recorder

cli --> gps
cli --> temp_sensor
cli --> conio

gps --> spi
temp_sensor --> i2c
conio --> serial
recorder --> fs
recorder --> cellular
@enduml
```

File: .gitignore (hunk 1 above)
Comment (emilybthorpe) [reply .gitignore:120]: Should I put the docs in a differently named folder then?

**Comment by emilybthorpe** [comment]
> I would suggest a diagram like this:
> 
> ```
> @startuml Service Diagram
> package "Device OS" {
>     [Cellular] as cellular
>     [I2C] as i2c
>     [Serial] as serial
>     [Filesystem] as fs
>     [SPI] as spi
> }
> 
> 
> package "Application" {
>     [CLI] as cli
>     [Ride] as ride
> }
> 
> package "Board Support Package" {
>     [Temp Sensor] as temp_sensor
>     [GPS] as gps
> }
> [ConIO] as conio
> [Recorder] as recorder
> 
> ride --> temp_sensor
> ride --> gps
> ride --> recorder
> 
> cli --> gps
> cli --> temp_sensor
> cli --> conio
> 
> gps --> spi
> temp_sensor --> i2c
> conio --> serial
> recorder --> fs
> recorder --> cellular
> @enduml
> ```

That seems good, I've added a diagram based on that. I'll add more sensors as those are added as well.

File: docs/architecture.md (hunk 2 above)
Comment (emilybthorpe) [reply docs/architecture.md]: Sounds good, I switched them back to # style

**Comment by emilybthorpe** [comment]
Is there other additional documentation components you'd suggest adding?

File: docs/application.puml
Comment (ntlhui) [inline docs/application.puml:1]: Note that you don't mention the LocationService, FLOG, NVRAM, LED, ChargeTask

File: .gitignore (hunk 1 above)
Comment (ntlhui) [reply .gitignore:120]: no

**Comment by ntlhui** [comment]
> Is there other additional documentation components you'd suggest adding?

Are there any other documents you think would have helped you get started on this project?

**Comment by emilybthorpe** [comment]
> > Is there other additional documentation components you'd suggest adding?
> 
> Are there any other documents you think would have helped you get started on this project?

I think the service diagram and architecture overview would have been really helpful, but I think those would be the main things.

**Review by ntlhui — APPROVED** [review]

**Pull request merged**